### PR TITLE
xfail kiwi image build on selinux enforcing hosts

### PR DESCRIPTION
**Pull request by dcermak**
[CI:TOXENVS] kiwi

### CODE REVIEW

**Review by m-dati — APPROVED** [review]

File: tests/test_kiwi.py [hunk 1]
```diff
@@ -77,8 +78,12 @@ def test_kiwi_create_image(
 
     assert container_per_test.connection.file("kiwi/build-tests").exists
 
-    kiwi_cmd = "kiwi-ng system build --description kiwi/build-tests/x86/leap/test-image-disk --set-repo obs://openSUSE:Leap:15.5/standard --target-dir /tmp/myimage"
-    container_per_test.connection.check_output(kiwi_cmd)
+    kiwi_cmd = "kiwi-ng system build --description kiwi/build-tests/x86/leap/test-image-disk --set-repo obs://openSUSE:Leap:15.5/standard --target-dir /var/tmp/myimage"
```
Comment (dirkmueller) [inline tests/test_kiwi.py:81]: why the change from `/tmp` to `/var/tmp` ?

File: tests/test_kiwi.py [hunk 2]
```diff
@@ -77,8 +78,12 @@ def test_kiwi_create_image(
...
+    res = container_per_test.connection.run_expect([0, 1], kiwi_cmd)
+    if res.rc == 1 and selinux_status() == "enforcing":
+        pytest.xfail(
+            "kiwi container fails to build an image on hosts in SELinux enforcing mode"
+        )
 
     container_per_test.connection.check_output(
         "kiwi-ng result list --target-dir=/tmp/myimage/"
```
Comment (dirkmueller) [inline tests/test_kiwi.py:89]: this is now broken

File: tests/test_kiwi.py (hunk 1 above)
Comment (dcermak) [reply tests/test_kiwi.py]: stupid oversight on my end, it's a leftover from some experimentation

File: tests/test_kiwi.py [hunk 3]
```diff
@@ -77,8 +78,12 @@ def test_kiwi_create_image(
 
     assert container_per_test.connection.file("kiwi/build-tests").exists
 
-    kiwi_cmd = "kiwi-ng system build --description kiwi/build-tests/x86/leap/test-image-disk --set-repo obs://openSUSE:Leap:15.5/standard --target-dir /tmp/myimage"
-    container_per_test.connection.check_output(kiwi_cmd)
+    kiwi_cmd = "kiwi-ng system build --description kiwi/build-tests/x86/leap/test-image-disk --set-repo obs://openSUSE:Leap:15.5/standard --target-dir /var/tmp/myimage"
+    res = container_per_test.connection.run_expect([0, 1], kiwi_cmd)
+    if res.rc == 1 and selinux_status() == "enforcing":
+        pytest.xfail(
+            "kiwi container fails to build an image on hosts in SELinux enforcing mode"
```
[the remark this answers is not in the document]
Comment (dcermak) [reply tests/test_kiwi.py:85]: Yes we could, but after working on the fips tests, I'm no longer convinced that a catch-all xfail is such a good idea. xfailing the whole function means that *any* failure during the test gets ignored, not just the failure that we're expecting.

[the remark this answers is not in the document]
Comment (alexandrevicenzi) [reply tests/test_kiwi.py:85]: what about a skipif selinux?

[the remark this answers is not in the document]
Comment (dirkmueller) [reply tests/test_kiwi.py:85]: I meant skip not xfail..?

[the remark this answers is not in the document]
Comment (dcermak) [reply tests/test_kiwi.py:85]: Isn't there hope that we might fix this eventually? If yes, then I'd say xfail, otherwise skip

**Review by alexandrevicenzi — APPROVED** [review]
